src/Altinn.App.Analyzers/MetadataAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Altinn.App.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class MetadataAnalyzer : DiagnosticAnalyzer
+{
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => Diagnostics.All;
+
+    public override void Initialize(AnalysisContext context)
+    {
+        // var configFlags = GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics;
+        var configFlags = GeneratedCodeAnalysisFlags.None;
+        context.ConfigureGeneratedCodeAnalysis(configFlags);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationAction(OnCompilation);
+        // System.Diagnostics.Debugger.Launch();
+    }
+
+    private void OnCompilation(CompilationAnalysisContext context)
+    {
+        if (
+            !context.Options.AnalyzerConfigOptionsProvider.GlobalOptions.TryGetValue(
+                "build_property.projectdir",
+                out var projectDir
+            )
+        )
+        {
+            context.ReportDiagnostic(Diagnostic.Create(Diagnostics.ProjectNotFound, Location.None));
+            return;
+        }
+
+#pragma warning disable RS1035 // Do not use APIs banned for analyzers
+        // On compilation end, it is not totally unreasonable to read a file
+        var file = Path.Combine(projectDir, "config/applicationmetadata.json");
+        if (!File.Exists(file))
+        {
+            context.ReportDiagnostic(Diagnostic.Create(Diagnostics.ApplicationMetadataFileNotFound, Location.None));
+            return;
+        }
+
+        try
+        {
+            var jsonText = File.ReadAllText(file);
+#pragma warning restore RS1035 // Do not use APIs banned for analyzers
+            var metadata = JObject.Parse(jsonText) ?? throw new Exception("Deserialization returned 'null'");
+            var dataTypes =
+                (metadata.GetValue("dataTypes") as JArray) ?? throw new JsonException("Failed to parse 'dataTypes'");
```

Comment:
Why Newtonsoft instead of system.text.json? I think we should try to standardise (or mostly I don't want to learn the library that seems to be moved away from).

Reply:
Simply because System.Text.Json [has a bunch of `netstandard2.0` dependencies](https://www.nuget.org/packages/System.Text.Json/9.0.0-preview.3.24172.9#dependencies-body-tab) whereas `Newtonsoft.Json` has none 😄 It's a bit of a hassle as transitive dependencies all the way down have to be copied. I gave it a shot when building this out but didn't want to invest a ton of time into it atm

Reply:
😭